src/index.js.flow

```diff
@@ -0,0 +1,32 @@
+// @flow strict
+
+export type Version = 0 | 1
+export type Codec = string
+export type Multihash = Buffer
+export type BaseEncodedString = string
+
+declare class CID<a> {
```

Comment:
@Gozala generics aren't a requirement for class defs are they? I don't see `a` being used in CID and am not sure how CID could be a generic container so does `<a>` mean something else here?

Reply:
> @Gozala generics aren't a requirement for class defs are they? 

Correct, they are not required 

> I don't see a being used in CID and am not sure how CID could be a generic container so does <a> mean something else here?

On my phone so can’t provide links but this is related to our other discussions on decode / encode. `CID<a>` implies here that it’s address for node `a` which in turn would allow `get(CID<a>):Promise<a>` without type parameter `a` it wouldn’t be possible to express that type of relationship.

Reply:
To be more specific: indeed `a` type param is not used here, but it’s there because users of this lib are expected to make use of that parameter

Reply:
Oh, wow, that's pretty funky. So tools that use flow definitions would follow that all the way through such a call chain and enforce that, or is this simply for the sake of descriptive completeness?

Reply:
Flow type-checker will indeed catch and report any miss-matches.

Reply:
BTW that’s also a weak point of TS (at least it was) as it throws away unused type parameters not used in the class / function definitions while flow keeps them for inference at call-sites. It may not seem that important but in practice it makes a huge difference